commit-graph.c

```diff
@@ -2129,6 +2161,9 @@ int write_commit_graph(struct object_directory *odb,
 		}
 	}
 
+	git_config_get_bool("commitGraph.writeGenerationData", &write_generation_data);
+	ctx->write_generation_data = write_generation_data;
+
```

Comment:
This doesn't need to be a config, as we don't really want users to toggle this. It could be a `GIT_TEST_COMMIT_GRAPH_NO_GDAT` environment variable. Then, your test could write with `GIT_TEST_COMMIT_GRAPH_NO_GDAT=1` and then check all of the reading cases.

A good way to check that everything still works properly would be to extend `t/t6600-test-reach.sh` to write one commit-graph file in this mode (say, `commit-graph-no-gdat` to match the `commit-graph-[full|half]` that already exist), then update `run_three_modes` to be `test_all_modes` and run one test using that `-no-gdat` file.

Reply:
Wow! That's so simple.

I spent a week wondering all the ways I should test reachability without GDAT chunk and the answer was as simple as to leverage the hard work that done by you and other contributers!

Thanks for the detailed response.